assets/stylesheets/application.css.scss

```
@@ -160,3 +162,19 @@ p {
     }
   }
 }
+
+// Bootstrap 5 suggested #000 as `color:` and hover `color:` as it violates
+// "WCAG 2.1 text color contrast ratio of 4.5:1" with `color: #fff`.
+// https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=12AEE2
+// https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=0F94C0
+$_color-for-btn-primary: #fff;
+.btn-primary {
+  @include button-variant(
+    $primary,
+    $primary,
+    $_color-for-btn-primary,
+    if($_color-for-btn-primary == $color-contrast-light, shade-color($primary, $btn-hover-bg-shade-amount), tint-color($primary, $btn-hover-bg-tint-amount)),
+    if($_color-for-btn-primary == $color-contrast-light, shade-color($primary, $btn-hover-border-shade-amount), tint-color($primary, $btn-hover-border-tint-amount)),
+    $_color-for-btn-primary
+  );
+}
```

The following might work well and be easier to understand:
```suggestion

// Bootstrap 5 would suggest `color:#000` for normal and `:hover` as it violates
// "WCAG 2.1 text color contrast ratio of 4.5:1" with `color: #fff`.
// https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=12AEE2 (2.56:1) (.btn-primary)
// https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=0F94C0 (3.48:1) (.btn-primary:hover)
//
// Upstream change: https://github.com/twbs/bootstrap/pull/30168
//
// cf. https://stackoverflow.com/questions/65170317/sass-scss-bootstrap-5-color-functions-not-working-well
$min-contrast-ratio: 2.55;
```

---

Applied in d6268af and works well. Resolving.